domain/modelmigration/state/state.go

```diff
@@ -66,3 +68,36 @@ FROM model`, ModelInfo{})
 
 	return result.ControllerUUID, nil
 }
+
+// GetAllInstanceIDs returns all instance IDs from the current model as
+// juju/collections set.
+func (s *State) GetAllInstanceIDs(ctx context.Context) (set.Strings, error) {
+
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Errorf("cannot get database to retrieve instance IDs: %w", err)
+	}
+
+	query := `
+SELECT &instanceID.instance_id
+FROM   machine_cloud_instance`
+	queryStmt, err := s.Prepare(query, instanceID{})
+	if err != nil {
+		return nil, errors.Errorf("preparing retrieve all instance IDs statement: %w", err)
+	}
+
+	var result []instanceID
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, queryStmt).GetAll(&result)
```

Comment:
I think you need to handle no rows error as a non error here ?

Reply:
No, no instances is not an error case. I added the test for that case.